### ISSUE TITLE
使用预训练模型不能复现Model zoo 中3.2 table中的acc

### ISSUE THREAD

**Issue by shengmingkai** [deepinsight/insightface#445]
作者您好！
我按照您的步骤和代码，
利用Model Zoo 中的“3.2 LResNet50E-IR,ArcFace@ms1m-refine-v1” 预训练模型，
利用Dataset Zoo中的“MS1M-IBUG (85K ids/3.8M images) [5,6]” 数据集，
运行命令参数为：
CUDA_VISIBLE_DEVICES='0,1' python -u /data/src/train_softmax.py --network r50 --per-batch-size 64 --loss-type 4 --ckpt 2  --margin-m 0.5 --pretrained '/data/ckpt/arcface/model,0' --data-dir /data/datasets/faces_ms1m_112x112 --prefix /data/ckpt/arcface/arcface-r50-s
进行训练。
但是我在日志中打印的acc的结果，在跑完3Epoch后：
lr-batch-epoch: 0.1 10824 3
testing verification..
(12000, 512)
infer time 17.748569
[lfw][100000]XNorm: 21.851122
[lfw][100000]Accuracy-Flip: 0.97883+-0.00573
testing verification..
(14000, 512)
infer time 20.685041
[cfp_fp][100000]XNorm: 17.821963
[cfp_fp][100000]Accuracy-Flip: 0.75800+-0.02343
testing verification..
(12000, 512)
infer time 17.510608
[agedb_30][100000]XNorm: 21.715151
[agedb_30][100000]Accuracy-Flip: 0.89000+-0.02486
saving 50
[100000]Accuracy-Highest: 0.90500

acc on lfw 基本在0.97 左右徘徊。
acc on cfp_fp 基本在0.75 左右徘徊。
acc on agedb_30 基本在0.89 左右徘徊。

请问一下，是不是我那一不对？还是有哪些细节没看到导致无法复现：
Mode Zoo 中 3.2中的table的 Performance:
Method | LFW(%) | CFP-FP(%) | AgeDB-30(%) | MegaFace(%)
Ours     | 99.80      | 92.74          | 97.76              | 97.64

希望得到您的回复！
谢谢！

**Comment by nttstar** [comment]
use 128*4 batch size

**Comment by shengmingkai** [comment]
> use 128*4 batch size

恩恩谢谢。
您好，还请问一下，是不是一定要batch size大到一定程度了才会拥有比较好的结果？（比方上面您说的128*4 即512）
我用的两张1080ti的显卡，单张11G显存。利用上面的参数、模型和数据集，当我用默认的batch size=128，，就会内存不足。那我如果我要想跑128*4的batch size这样的硬件估计是不能的了么？
我也尝试了用“CASIA-Webfac”这个数据集跑batch size=128 的，但是这个效果更不好，在4个Epoch后，三个acc基本都会在0.5 左右了 。

**Comment by shengmingkai** [comment]
@nttstar 
恩恩，谢谢。
您好，想再请问一下，是不是一定要batch size大到一定程度了才会拥有比较好的结果？（比方上面您说的128*4 即512）
我用的两张1080ti的显卡，单张11G显存。利用上面的参数、模型和数据集，当我用默认的batch size=128，，就会内存不足。那我如果我要想跑128*4的batch size这样的硬件估计是不能的了？对硬件会有要求？如果我这样的硬件如果想跑出重现table的结果我可以尝试用什么样的数据集和参数？
我也尝试了用“CASIA-Webfac”这个数据集跑batch size=128 的，但是这个效果更不好，在4个Epoch后，三个acc基本都会在0.5 左右了 
希望得到回复！
谢谢！

**Comment by shengmingkai** [comment]
@nttstar 
恩恩，谢谢您。
我在保存的ckpt文件上，调整了学习率后，acc上去了。应该是学习率的过大导致的无法收敛。
谢谢！

**Comment by wting861006** [comment]
@shengmingkai 我的也是两张1080ti显卡，请问你跑一个epoch要多长时间？

**Comment by shengmingkai** [comment]
> @shengmingkai 我的也是两张1080ti显卡，请问你跑一个epoch要多长时间？

当时在ArcFace@ms1m-refine-v1，上面的参数，一个epcoh大概几个小时吧。我记得不是很清楚了。

**Comment by Po-Shin** [comment]
@shengmingkai May I ask what are the accuracies you got during the experiment?

Thanks!

**Comment by shengmingkai** [comment]
> @shengmingkai May I ask what are the accuracies you got during the experiment?
> 
> Thanks!

only  0.98+  @lfw。

**Comment by wting861006** [comment]
我是4张1080ti做的训练，我的训练结果到这个样子基本上就不会下降了。请问我是不是可以停止训练了？
>>python -u train.py --network r100 --loss arcface --dataset emore --per-batch-size 50 --pretrained D:\insightface\models\r100-arcface-emore\model --pretrained-epoch 1 --models-root D:\insightface\models --lr 0.0001

lr-batch-epoch: 1e-06 5999 0
testing verification..
(12000, 512)
infer time 30.214935999999994
[lfw][6000]XNorm: 22.268995
[lfw][6000]Accuracy-Flip: 0.99767+-0.00300
testing verification..
(14000, 512)
infer time 32.406338999999996
[cfp_fp][6000]XNorm: 19.892518
[cfp_fp][6000]Accuracy-Flip: 0.92271+-0.01446
testing verification..
(12000, 512)
infer time 27.741421999999993
[agedb_30][6000]XNorm: 22.853581
[agedb_30][6000]Accuracy-Flip: 0.97650+-0.00787
testing verification..
(14000, 512)
infer time 32.28423
[cfp_ff][6000]XNorm: 21.346916
[cfp_ff][6000]Accuracy-Flip: 0.99786+-0.00214
[6000]Accuracy-Highest: 0.99814
INFO:root:Epoch[0] Batch [5980-6000]    Speed: 25.27 samples/sec        acc=0.707750    lossvalue=1.932763
INFO:root:Epoch[0] Batch [6000-6020]    Speed: 197.73 samples/sec       acc=0.712250    lossvalue=1.844488
INFO:root:Epoch[0] Batch [6020-6040]    Speed: 195.41 samples/sec       acc=0.721250    lossvalue=1.776921
INFO:root:Epoch[0] Batch [6040-6060]    Speed: 196.80 samples/sec       acc=0.706250    lossvalue=1.962833
INFO:root:Epoch[0] Batch [6060-6080]    Speed: 197.21 samples/sec       acc=0.708750    lossvalue=1.906868
INFO:root:Epoch[0] Batch [6080-6100]    Speed: 200.57 samples/sec       acc=0.711250    lossvalue=1.885361
INFO:root:Epoch[0] Batch [6100-6120]    Speed: 184.77 samples/sec       acc=0.723500    lossvalue=1.785582
INFO:root:Epoch[0] Batch [6120-6140]    Speed: 197.78 samples/sec       acc=0.718750    lossvalue=1.862185
INFO:root:Epoch[0] Batch [6140-6160]    Speed: 198.07 samples/sec       acc=0.713750    lossvalue=1.762263
INFO:root:Epoch[0] Batch [6160-6180]    Speed: 198.57 samples/sec       acc=0.707000    lossvalue=1.912943
INFO:root:Epoch[0] Batch [6180-6200]    Speed: 198.69 samples/sec       acc=0.718750    lossvalue=1.791566
INFO:root:Epoch[0] Batch [6200-6220]    Speed: 196.06 samples/sec       acc=0.714250    lossvalue=1.836867
INFO:root:Epoch[0] Batch [6220-6240]    Speed: 198.66 samples/sec       acc=0.707750    lossvalue=1.904006
INFO:root:Epoch[0] Batch [6240-6260]    Speed: 193.76 samples/sec       acc=0.730500    lossvalue=1.787488
INFO:root:Epoch[0] Batch [6260-6280]    Speed: 198.03 samples/sec       acc=0.709500    lossvalue=1.897590
INFO:root:Epoch[0] Batch [6280-6300]    Speed: 198.30 samples/sec       acc=0.700000    lossvalue=1.913924

**Comment by stormand** [comment]
> use 128*4 batch size

请问，128*4 和 64*8的效果是一样的吗? 我单张卡也无法跑到128的batchsize
